airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java

```diff
@@ -81,15 +83,15 @@ public AirbyteMessageConsumer create(final JsonNode config,
 
   // TODO Commenting this out for now since it slows down syncs
   private CheckedConsumer<AirbyteStreamNameNamespacePair, Exception> incrementalTypingAndDedupingStreamConsumer(final TyperDeduper typerDeduper) {
-    // final TypeAndDedupeOperationValve valve = new TypeAndDedupeOperationValve();
+    final TypeAndDedupeOperationValve valve = new TypeAndDedupeOperationValve();
```

Comment:
chatted about this a bit offline, but generally concerned about the performance impact of enabling this not asynchronously. Will run some tests locally to confirm

Reply:
ran a test locally - source postgres to bq gcs staging w/ 30 streams all full refresh, commenting out lines 91-94 sync time was ~12.5min leaving this as is and running the same sync is ~20 minutes at the very least we should comment out these lines otherwise I can add the commit we talked about @edgao to async this for bq

Reply:
+1 on commenting this out again. for now I think async destinations should get the incremental T+D since it's relatively free, but maybe we just skip incremental T+D for non-async destinations

destination-bigquery async rollout is probably happening soon, so we can uncomment it once that happens

Reply:
fixed.

Reply:
re: commenting out those lines
ran another test locally where I uncommented these lines now that `typeAndDedupe` for individual streams is using the type-and-dedupe thread pool, and it has similar performance to running this without using the separate thread pool.
TL;dr stays commented out for now until BigQuery GCS is asyncified because it slows down the overall sync